internal/wire/version_negotiation.go

```diff
@@ -7,17 +7,33 @@ import (
 	"github.com/lucas-clemente/quic-go/internal/utils"
 )
 
-// ComposeVersionNegotiation composes a Version Negotiation Packet
-// TODO(894): implement the IETF draft format of Version Negotiation Packets
-func ComposeVersionNegotiation(connectionID protocol.ConnectionID, versions []protocol.VersionNumber) []byte {
+// ComposeGQUICVersionNegotiation composes a Version Negotiation Packet for gQUIC
+func ComposeGQUICVersionNegotiation(connID protocol.ConnectionID, versions []protocol.VersionNumber) []byte {
 	fullReply := &bytes.Buffer{}
 	ph := Header{
-		ConnectionID: connectionID,
+		ConnectionID: connID,
 		PacketNumber: 1,
 		VersionFlag:  true,
 	}
-	err := ph.writePublicHeader(fullReply, protocol.PerspectiveServer, protocol.VersionWhatever)
-	if err != nil {
+	if err := ph.writePublicHeader(fullReply, protocol.PerspectiveServer, protocol.VersionWhatever); err != nil {
+		utils.Errorf("error composing version negotiation packet: %s", err.Error())
```

Comment:
Is a `return nil` missing here?

Reply:
Yes, but this was an existing issue. `writePublicHeader` shouldn't fail when writing to a `bytes.Buffer`, unless given the wrong input values.